include/Partitioning/EBV.hpp

```diff
@@ -0,0 +1,179 @@
+            {
+
+                int usleep_time = 2;
+                while (!u_record->getLock())
+                {
+                    std::this_thread::sleep_for(std::chrono::microseconds(usleep_time));
+                    usleep_time = (int)pow(usleep_time, 2);
+                }
+                usleep_time = 2;
+                while (!v_record->getLock())
+                {
+                    std::this_thread::sleep_for(std::chrono::microseconds(usleep_time));
+                    usleep_time = (int)pow(usleep_time, 2);
+
+                    if (usleep_time > GLOBALS.SLEEP_LIMIT)
+                    {
+                        u_record->releaseLock();
+                        performStep(e, state);
+                        return;
+                    } //TO AVOID DEADLOCK
+                }
+                locks_taken = true;
+            }
+            //*** LOCK TAKEN
+            int machine_id = -1;
+
+            //*** COMPUTE SCORES, FIND MIN SCORE, AND COMPUTE CANDIDATES PARTITIONS
+            std::map<int, double> eva;
+            auto u_partition = u_record->getPartitions();
+            auto v_partition = v_record->getPartitions();
+            auto optimalEdgesNumber = edgeCardinality / P;
+            auto optimalVerticesNumber = vertexCardinality / P;
+            for (int i = 0; i < P; i++)
+            {
+                eva[i] = 0;
+                if (u_partition.empty() || u_partition.find(i) == u_partition.end())
+                {
+                    eva[i] += 1;
+                }
+                if (v_partition.empty() || v_partition.find(i) == v_partition.end())
+                {
+                    eva[i] += 1;
+                }
+                eva[i] += alpha * (state.getMachineLoad(i) / optimalEdgesNumber) + beta * (state.getMachineLoadVertices(i) / optimalVerticesNumber);
+            }
+            //find min between eva
+            double min = eva.at(0);
+            machine_id = 0;
+            for (auto &it : eva)
+            {
+                if (it.second < min)
+                {
+                    min = it.second;
+                    machine_id = it.first;
+                }
+            }
+            try
+            {
+                CoordinatedPartitionState<T> &cord_state = dynamic_cast<CoordinatedPartitionState<T> &>(state);
+                //NEW UPDATE RECORDS RULE TO UPFDATE THE SIZE OF THE PARTITIONS EXPRESSED AS THE NUMBER OF VERTICES THEY CONTAINS
+                if (!u_record->hasReplicaInPartition(machine_id))
+                {
+                    u_record->addPartition(machine_id);
+                    cord_state.incrementMachineLoadVertices(machine_id);
+                }
+                if (!v_record->hasReplicaInPartition(machine_id))
+                {
+                    v_record->addPartition(machine_id);
+                    cord_state.incrementMachineLoadVertices(machine_id);
+                }
+            }
+            catch (std::bad_cast& e)
```

Comment:
The excpetion could be passed as "const" ? Aesthetic change, slightly more secure.

Reply:
resolved in 2ad39b9c3a66666490c5a3ab3807cfbed78547f6